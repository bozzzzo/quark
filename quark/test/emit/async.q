
class Later extends Async<String> {
    void callback(String result) {
        print("later: " + result);
    }
    void errback(String failure) {
        print("later failure: " + failure);
    }
}

class Requestbin extends Async<String> {
    int done;
    Requestbin() {
        self.done = 42;
    }
    void callback(String result) {
        done = done + 13;
        print("requestbin: " + result + " done=" + done.toString());
    }
    void errback(String failure) {
        done = done + 1;
        print("requestbin failure: " + failure);
    }
}

void main() {
    Async<String> l = new Later();
    l.callback("expected");
    l.errback("expected");
    Requestbin r = new Requestbin();
    url_get_async("http://127.0.0.1:9998/yay!", r);
    url_get_async("http://127.0.0.1:9999/yay!", r);
}
