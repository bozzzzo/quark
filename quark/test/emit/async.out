later: expected
later failure: expected
requestbin failure: error
requestbin: yay! done=56
